### PR TITLE
make kp.Store into an interface, with consulStore implementing that i…

### PR DESCRIPTION
**Pull request by mpuncel**
…nterface

### CODE REVIEW

**Comment by anthonybishopric** [comment]
:+1:

File: pkg/kp/kv.go [hunk 1]
```diff
@@ -17,12 +17,19 @@ type ManifestResult struct {
...
+}
+
+type consulStore struct {
 	client *api.Client
 }
 
-func NewStore(opts Options) *Store {
-	return &Store{client: NewConsulClient(opts)}
+func NewStore(opts Options) Store {
+	return &consulStore{client: NewConsulClient(opts)}
```
Comment (bwester) [inline pkg/kp/kv.go]: I don't think you need to return a pointer here.

**Comment by bwester** [comment]
From the CI build failures, there are some more call sites to convert, but otherwise this is looking good.

**Comment by bwester** [comment]
Github has problems displaying your commit message.  Is it a problem with your commit itself, or does Github just not like long lines?

**Comment by mpuncel** [comment]
the commit message just wrapped into the description. The full commit message is "make kp.Store into an interface, with consulStore implementing that interface"

**Comment by petertseng** [comment]
github is cutting off at 70 chars (counting the trailing "…" as one character)

**Comment by bwester** [comment]
:ship:

**Pull request merged**